src/hiveboard_bridge/include/hiveboard_bridge/TCPServer.h

```diff
@@ -0,0 +1,68 @@
+#ifndef CATKIN_ROS_TCPSERVER_H
+#define CATKIN_ROS_TCPSERVER_H
+
+#include "ITCPServerMonitor.h"
+#include "TCPServerMonitor.h"
+#include "ros/ros.h"
+#include <cstdint>
+#include <cstring>
+#include <memory>
+#include <netinet/in.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#define TCP_BUFFER_LENGTH UINT16_MAX
+
+class TCPServer {
+  public:
+    /**
+     * Construct a TCPServer
+     * @param port Port to open the server to
+     */
+    TCPServer(int port);
+    ~TCPServer();
+
+    /**
+     * Start listening for incoming client connections and accept one client. This is blocking until
+     * a client connects.
+     */
+    void listen();
+
+    /**
+     * Read data from socket.
+     * @param buff The buffer on which the incoming data will be written.
+     * @param length The length of the provided buffer.
+     * @param blocking Whether the call should block until some data is available.
+     * @return The length of the data read.
+     */
+    int read(char* buff, const uint16_t length, bool blocking);
+
+    /**
+     * Send some data to the client.
+     * @param buff The buffer containing the data to send.
+     * @param length The length of the data on the buffer.
+     */
+    void send(char* buff, const uint16_t length);
+
+    /**
+     * Terminate a client connection.
+     */
+    void close();
+
+  private:
+    int m_serverFd, m_clientFd, m_port;
+    int m_addressLength;
+    struct sockaddr_in m_address;
+    char m_buffer[TCP_BUFFER_LENGTH];
```

Comment:
Isn't it a bit abusive to allocate an array of 65kb? I know we are afraid of heap on HiveMind and Hiveconnect, but you can use std::string here if you want, don't be affraid of using heap since you are on linux.

You could even pass a std::string instead of a char* and lenght for send and receive methods for example, though it's really fine the way you did it too (I'm not asking you to change it, it's just as an example of std::string uses).

Reply:
I spoke with @cquesnel and we had this idea that we should maybe limit the maximum size of the data that can be sent at once via TCP to 65k. That way the recieve buffer in the HiveMind could be 65k maximum (which seems reasonable; not too big, not too small). 
I just thought that adopting the same strategy here might force us to seperate big chunks of data in bits of 65k (and maybe alleviate some problems in the long run).